src/ir_York.h

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 Daniele Gobbetti
+
+/// @file
+/// @brief Support for the York AC protocol (remote GRYLH2A)
+
+// Supports:
+//   Brand: York,  Model: MHH07P17
+
+#ifndef IR_YORK_H_
+#define IR_YORK_H_
+
+#define __STDC_LIMIT_MACROS
+#include <stdint.h>
+#ifndef UNIT_TEST
+#include <Arduino.h>
+#endif
+#include "IRremoteESP8266.h"
+#include "IRsend.h"
+#ifdef UNIT_TEST
+#include "IRsend_test.h"
+#endif
+
+
+/// Native representation of a York A/C message.
+union YorkProtocol{
+  uint8_t raw[kYorkStateLength];  ///< The state of the IR remote.
+  struct {
+    // byte 0-5
+    uint8_t preamble[6];  // unknown, fixed 0x08, 0x10, 0x07, 0x02, 0x40, 0x08
+    // byte 6
+    uint8_t Key1     :4;  // key pressed on the remote: 1 power,  2 temp up, 3
+                          // temp down...
+    uint8_t Key2     :4;  // only set when setting ontime/offtime:
+                          // Key1 value is 0x6 (enter key) and Key2 is 0x3 for
+                          // "start" and 0x2 for "stop"
+    // byte 7
+    uint8_t Fan      :4;  // Fan speed: 1 low, 2 mid, 3 max, 8 auto
+    uint8_t Power    :1;  // main unit power: 1 on, 0 off
+    uint8_t          :3;
+    // byte 8
+    uint8_t Mode     :4;  // 1 heat, 2 cool, 3 dry, 4 fan, 8 auto
+    uint8_t          :4;
+    // byte 9
+    uint8_t          :2;
+    uint8_t Temp     :6;  // Degrees Celsius
+    // byte 10
+    uint8_t OffTimer :8;  // Power off time: 10s of minutes from now
+    // byte 11
+    uint8_t OnTimer  :8;  // Power on time: 10s of minutes from now
+    // byte 12
+    uint8_t          :8;  // unknown, normalyy 0x00, 0x08 when ontime set,
+                          // 0x88 if both on and offtime set (could mean "auto
+                          // power")
```

Comment:
Suggestion:
```c++
  // Byte 12
  uint8_t             :3;
  uint8_t OnTimerSet  :1;
  uint8_t             :3;
  uint8_t OffTimerSet :1;
```

That should allow you to easily test if they are set or not. Based on your observations of `0x00` all off, `0x08` On timer set. `0x80` Off timer set, `0x88` both set.

Reply:
My assumptions about those specific bytes were wrong, I'm back to the drawing board. See [here](https://docs.google.com/spreadsheets/d/1PsI8KSKzd8HQQeZUcCiCktq7l3xWww1AtdCdbUrWmfg/edit?usp=sharing)